### ISSUE TITLE
Overzealous SC2086

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1703]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bash

declare -a ARGV
ARGV=()
while (($#))
do
    case "$1" in
        -dbtrace )
            export DBIC_TRACE=1
            echo DBIC_TRACE=$DBIC_TRACE
            ;;
        -d | -d:ptkdb )
            export PORTAL_DEBUG=$1
            echo PORTAL_DEBUG=$PORTAL_DEBUG ## Line 14
            ;;
        * )
            ARGV+=("$1")
            ;;
    esac
    shift
done

```

#### Here's what shellcheck currently says:
```
Line 14	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

Nothing. Via the case statement, `$1` must be `-d` or `-d:ptkdb`, neither of which have spaces in them. Therefore the usage of `$PORTAL_DEBUG` at line 14 cannot have spaces either. Can shellcheck do this sort of transitive static analysis, or am I asking too much?
